clusterloader2/cmd/clusterloader.go

```diff
@@ -307,22 +305,51 @@ func main() {
 
 	testReporter := test.CreateSimpleReporter(path.Join(clusterLoaderConfig.ReportDir, "junit.xml"), "ClusterLoaderV2")
 	testReporter.BeginTestSuite()
+
+	var contexts []test.Context
+
 	if testSuiteConfigPath != "" {
 		testSuite, err := config.LoadTestSuite(testSuiteConfigPath)
 		if err != nil {
 			klog.Exitf("Error while reading test suite: %v", err)
 		}
+
 		for i := range testSuite {
-			clusterLoaderConfig.TestScenario = testSuite[i]
-			runSingleTest(f, prometheusFramework, testReporter)
+			testScenario := testSuite[i]
+			ctx, errList := test.CreateTestContext(f, prometheusFramework, &clusterLoaderConfig, testReporter, &testScenario)
+			if !errList.IsEmpty() {
+				klog.Exitf("Test context creation failed: %s", errList.String())
+			}
+			testConfig, errList := test.CompileTestConfig(ctx)
+			if !errList.IsEmpty() {
+				klog.Exitf("Test compliation failed: %s", errList.String())
+			}
+			ctx.SetTestConfig(testConfig)
+			contexts = append(contexts, ctx)
 		}
 	} else {
 		for i := range testConfigPaths {
-			clusterLoaderConfig.TestScenario.ConfigPath = testConfigPaths[i]
-			clusterLoaderConfig.TestScenario.OverridePaths = testOverridePaths
-			runSingleTest(f, prometheusFramework, testReporter)
+			testScenario := api.TestScenario{
+				ConfigPath:    testConfigPaths[i],
+				OverridePaths: []string{},
```

Comment:
We should set it to the `clusterLoaderConfig.OverridePaths`

Reply:
Not sure if this would be confusing because later I end up doing:
```go
mapping, err := LoadTestOverrides(clusterLoaderConfig.OverridePaths, testOverridePaths)
```